src/utils/cryptography.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Cryptographic helpers.
+ */
+import nacl from "tweetnacl";
+import util from "tweetnacl-util";
+
+/**
+ * Encrypt a Blob (File) with a new secret key.
+ *
+ * This is currently one-shot: the Blob should fit in memory.
+ * For streaming encryption, we should look at nacl-stream-js or nacl-blob.
+ *
+ * @see https://github.com/dchest/nacl-stream-js
+ * @see https://github.com/bcomnes/nacl-blob
+ */
+export async function encryptBlob(
+  plainBlob: Blob
+): Promise<{
+  cipherBlob: Blob;
+  keyBase64: string;
+  nonceBase64: string;
```

Comment:
@PiDelport  These nonces are random right? Is there a reason we might need to store those in session storage, or do we return it just for the display at this point?

Reply:
Yes, they're random. The nonce is considered public, and would usually be transmitted and stored along with the encrypted blob, but we could also just store or transmit it alongside the secret key.

Reply:
http://cr.yp.to/highspeed/coolnacl-20120725.pdf#page=5 has some discussion about nonces in the context of NaCl, and how they might fit into higher-level protocols.